presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -5301,6 +5301,93 @@ public void testUnsupportedCsvTable()
     }
 
+    @Test
+    public void testCreateMaterializedView()
+    {
+        computeActual("CREATE TABLE test_customer_base WITH (partitioned_by = ARRAY['nationkey']) AS SELECT custkey, name, address, nationkey FROM customer LIMIT 10");
+        computeActual("CREATE TABLE test_orders_base WITH (partitioned_by = ARRAY['orderstatus']) AS SELECT orderkey, custkey, totalprice, orderstatus FROM orders LIMIT 10");
+
+        // Test successful create
+        assertUpdate("CREATE MATERIALIZED VIEW test_customer_mv WITH (partitioned_by = ARRAY['nationkey']) AS SELECT name, nationkey FROM test_customer_base");
+        assertTrue(getQueryRunner().tableExists(getSession(), "test_customer_mv"));
+        assertTableColumnNames("test_customer_mv", "name", "nationkey");
+
+        // Test if exists
+        assertQueryFails(
+                "CREATE MATERIALIZED VIEW test_customer_mv AS SELECT name FROM test_customer_base",
+                format(".* Materialized view '%s.%s.test_customer_mv' already exists", getSession().getCatalog().get(), getSession().getSchema().get()));
+        assertQuerySucceeds("CREATE MATERIALIZED VIEW IF NOT EXISTS test_customer_mv AS SELECT name FROM test_customer_base");
+
+        // Test partition mapping
+        assertQueryFails(
+                "CREATE MATERIALIZED VIEW test_customer_mv_no_partition AS SELECT name FROM test_customer_base",
+                ".*Unpartitioned materialized view is not supported.");
+        assertQueryFails(
+                "CREATE MATERIALIZED VIEW test_customer_mv_no_direct_partition_mapping WITH (partitioned_by = ARRAY['nationkey']) AS SELECT name, CAST(nationkey AS BIGINT) AS nationkey FROM test_customer_base",
+                format(".*Materialized view %s.test_customer_mv_no_direct_partition_mapping must have at least one partition directly defined by a base table partition.", getSession().getSchema().get()));
+
+        // Test nested
+        assertQueryFails(
+                "CREATE MATERIALIZED VIEW test_customer_nested_mv WITH (partitioned_by = ARRAY['nationkey']) AS SELECT name, nationkey FROM test_customer_mv",
+                format(".*CreateMaterializedView on a materialized view %s.%s.test_customer_mv is not supported.", getSession().getCatalog().get(), getSession().getSchema().get()));
+
+        // Test query shape
+        assertUpdate("CREATE MATERIALIZED VIEW test_customer_agg_mv WITH (partitioned_by = ARRAY['nationkey']) AS SELECT COUNT(DISTINCT name) AS num, nationkey FROM (SELECT name, nationkey FROM test_customer_base GROUP BY 1, 2) a GROUP BY nationkey");
+        assertUpdate("CREATE MATERIALIZED VIEW test_region_nation_join_mv WITH (partitioned_by = ARRAY['orderstatus', 'nationkey']) AS SELECT orders.totalprice, orders.orderstatus, customer.nationkey FROM test_customer_base customer JOIN test_orders_base orders ON orders.custkey = customer.custkey");
```

Comment:
Can we add a test where a join query will fail? Especially more tests to cover partition column mapping.